src/navigation/path.cpp

```diff
@@ -261,14 +360,20 @@ double Path::speedAlongPath(double traveledDistance) const {
     }
 
     // Dampen speed in end of path
-    // Note: this leads to problems when the full length of the path is really big
-    double closeUpDistance = 2.0 * _end.node()->boundingSphere();
-    if (closeUpDistance < Epsilon) { // zero bounding sphere
+    double closeUpDistance = closeUpDistanceFactor * _end.node()->boundingSphere();
+    if (closeUpDistance < LengthEpsilon) { // zero bounding sphere
         closeUpDistance = glm::distance(_end.position(), endNodePos);
     }
 
     if (traveledDistance > (pathLength() - closeUpDistance)) {
-        const double remainingDistance = pathLength() - traveledDistance;
+        double remainingDistance = 0.0;
+        if (_type == Type::Linear) {
+            remainingDistance = glm::distance(_prevPose.position,_end.position());
+        }
+        else {
+            remainingDistance = pathLength() - traveledDistance;
+        }
+
         speed *= remainingDistance / closeUpDistance + 0.01;
```

Comment:
Why the 0.01 here?

Reply:
The reason is that we want to prevent multiplying the speed with exactly 0.0, as this would result in a zero speed and no movement/displacement. 

Refactored the speed part of the code a bit, so hopefully, now it's clearer